workers/unity/Packages/com.improbable.gdk.core/Systems/CleanTemporaryComponentsSystem.cs

```diff
@@ -20,24 +19,25 @@ protected override void OnCreateManager()
         {
             base.OnCreateManager();
 
-            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            // Find all components with the RemoveAtEndOfTick attribute
+            foreach (var type in ReflectionUtility.GetNonAbstractTypes(typeof(IComponentData)))
             {
-                // Find all components with the RemoveAtEndOfTick attribute
-                foreach (var type in assembly.GetTypes())
+                if (type.GetCustomAttribute<RemoveAtEndOfTickAttribute>(false) == null)
                 {
-                    if (type.GetCustomAttribute<RemoveAtEndOfTickAttribute>(false) == null)
-                    {
-                        continue;
-                    }
+                    continue;
+                }
 
-                    if (!typeof(IComponentData).IsAssignableFrom(type)
-                        && !typeof(ISharedComponentData).IsAssignableFrom(type))
-                    {
-                        continue;
-                    }
+                componentGroupsToRemove.Add((GetComponentGroup(ComponentType.ReadOnly(type)), type));
+            }
 
-                    componentGroupsToRemove.Add((GetComponentGroup(ComponentType.ReadOnly(type)), type));
+            foreach (var type in ReflectionUtility.GetNonAbstractTypes(typeof(ISharedComponentData)))
+            {
+                if (type.GetCustomAttribute<RemoveAtEndOfTickAttribute>(false) == null)
```

Comment:
Why is this explicitly ignoring inheritance?

Reply:
Mistake. This one is supposed to be allowed